BondSource.cpp

```diff
@@ -0,0 +1,165 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2014 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+#include <ibamr/config.h>
+
+#include <ibamr/app_namespaces.h>
+
+#include <HierarchyDataOpsManager.h>
+#include <SAMRAI_config.h>
+#include <math.h>
+// local include
+#include "BondSource.h"
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+
+BondSource::BondSource(SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> phi_u_var,
+                    SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> phi_a_var,
+                    SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM>> w_var,
```

Comment:
Take a look at the modifications I made to the platelet sources for the wall sites. Wall sites are not controlled by the advection diffusion integrator, we have to get them in a different way.
https://github.com/abarret/clotting_model/blob/0251592732b153423aab269b6a2d2159b2d282f1/PlateletSource.h#L89